### PR TITLE
containerize service

### PR DESCRIPTION
**Pull request by ebuckley**
## TODO

- [ ] Push docker image to govtnz
- [ ] Integrate with existing deployment pipeline for data-govt
- [ ] Update docker docs for deployment/push to docker

## Changes
- update dockerfile to install gpg
- add entrypoint script for running fetch/decrypt/publish in one run
- update docs
- only load a `.env` file if it exists in the decrypt/fetch scripts
- add some docs

### CODE REVIEW

**Comment by camfindlay** [comment]
Does this branch include the commits from `getting-started` branch... there was some script improvements alongside the container work we'll need to ensure are included here too @ebuckley

**Comment by camfindlay** [comment]
Just looking over and see that it's all including the other commits.
Good to merge.

File: Dockerfile [hunk 1]
```diff
@@ -2,4 +2,6 @@ FROM python:2.7-slim
 COPY . /app
 WORKDIR /app
 RUN pip install --trusted-host pypi.python.org -r requirements.txt
-RUN apt-get update; apt-get install ssh-client wget -y
+RUN apt-get update; apt-get install ssh-client wget gpg -y --no-install-recommends
```
Comment (tfentonz) [inline Dockerfile]: Worthwhile cleaning up the apt cache at end of this RUN command `&& rm -rf /var/lib/apt/lists/*`.
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run

Comment (camfindlay) [reply Dockerfile]: Good catch

File: docker-cmd.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+if [ -z "$SFTP_HOST" ]; then
+    exit 1
+fi
+
+if [ -z "$SFTP_DIR" ]; then
+    exit 1
+fi
+
+if [ ! -f "$DECRYPTION_KEY" ]; then
```
Comment (camfindlay) [inline docker-cmd.sh:39]: What if we arn't decrypting the data we are picking up?

File: docker-cmd.sh [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    echo "Decryption key does not exist $DECRYPTION_KEY"
+    exit 1
+fi
+
+echo "Import the private key for decrypting data"
+gpg --import "$DECRYPTION_KEY"
+
+echo "Starting the data loading pipeline"
+./fetch.sh && ./decrypt.sh && python publish.py
```
Comment (camfindlay) [inline docker-cmd.sh]: See above - need to cater for a "fetch and publish" pipeline also.

File: docker-cmd.sh (hunk 2 above)
Comment (camfindlay) [reply docker-cmd.sh:39]: You could use this check to decide if the intention is to run a decrypt process or not.

**Comment by camfindlay** [comment]
LGTM

**Pull request merged**